### PR TITLE
Add Invisible Character for hiding URL

### PR DESCRIPTION
**Pull request by OmegaSunkey**
This PR adds an option to NitroSpoof to hide the Emote URL using an invisible character trick and markdown capabilities.

### CODE REVIEW

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt [hunk 1]
```diff
@@ -71,6 +76,11 @@ class NitroSpoof : Plugin() {
         if (emoteSize != null) {
             finalUrl += "&size=${emoteSize}"
         }
+
+        if(settings.getBool("emptyChar", false)) {
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt:80]: Replace that code with the one below
```kt
if (settings.getBool("emptyChar", false) {
	finalUrl = EMPTY_CHAR + '(' + finalUrl + ')'
}
```

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt [hunk 2]
```diff
@@ -18,6 +20,7 @@ class PluginSettings(
         super.onViewBound(view)
 
         val context = requireContext()
+        val warning = "This allows you to embed your emote without the URL being present. Some servers do not allow empty character."
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:23]: There's no need to have this warning as a separate variable, inline it

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt [hunk 3]
```diff
@@ -58,7 +59,11 @@ class NitroSpoof : Plugin() {
             return
         }
 
-        var finalUrl = "https://cdn.discordapp.com/emojis/"
+        var finalUrl = (if (settings.getBool("emptyChar", false)) {
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt:62]: remove all of this as we do everything necessary below

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt [hunk 4]
```diff
@@ -37,12 +38,12 @@ class NitroSpoof : Plugin() {
             ModelEmojiCustom::class.java.getDeclaredMethod("isAvailable"),
             InsteadHook { true }
         )
-        commands.registerCommand("freenitroll", "Get free nitro (this is a troll)") {
+        /* commands.registerCommand("freenitroll", "Get free nitro (this is a troll)") {
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt:41]: why was this commented out?

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt [hunk 5]
```diff
@@ -37,7 +40,19 @@ class PluginSettings(
             }
         }
 
+        val emptyToggle = Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, "Enable Empty Character", warning)
+        emptyToggle.isChecked = settingsAPI.getBool("emptyChar", false)
+        emptyToggle.setOnCheckedListener{
+        	c: Boolean? ->
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:46]: the arguments should be placed on the same line as the lambda begins, also add spaces before brackets and parenthesis please

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt [hunk 6]
```diff
@@ -37,7 +40,19 @@ class PluginSettings(
             }
         }
 
+        val emptyToggle = Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, "Enable Empty Character", warning)
+        emptyToggle.isChecked = settingsAPI.getBool("emptyChar", false)
+        emptyToggle.setOnCheckedListener{
+        	c: Boolean? -> 
+        	if(PluginManager.isPluginEnabled("MoreHighlight")) settingsAPI.setBool("emptyChar", c!!) else {
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:47]: place the else block on a separate line

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/remove.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+remove now
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/remove.txt:1]: what is this?

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt [hunk 8]
```diff
@@ -2,3 +2,4 @@ package com.xinto.aliuplugins.nitrospoof
 
 const val EMOTE_SIZE_DEFAULT = "48"
 const val EMOTE_SIZE_KEY = "emote_size"
+const val EMPTY_CHAR = "[‎ ]"
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:5]: probably better to inline this

**Review by X1nto — CHANGES_REQUESTED** [review]

**Comment by X1nto** [comment]
> Please wait until I add server blacklist, as not even the Aliucord Server allows you to use this.

Ah, didn't see this comment before reviewing. Use the Draft PR feature next time.

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt [hunk 9]
```diff
@@ -1,4 +1,9 @@
 package com.xinto.aliuplugins.nitrospoof
 
+import com.aliucord.Constants.ALIUCORD_GUILD_ID
+
 const val EMOTE_SIZE_DEFAULT = "48"
 const val EMOTE_SIZE_KEY = "emote_size"
+const val EMPTY_CHAR = "[‎ ]"
+val permBlacklist = mapOf(ALIUCORD_GUILD_ID to "Aliucord")
+val servBlacklist = mutableMapOf(1015931589865246730 to "Vendetta")
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:9]: Why is this a separate list? Also why is the Vendetta server blacklisted?

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt (hunk 4 above)
Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt:67]: so people dont delete their nitro

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/remove.txt (hunk 7 above)
Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/remove.txt:1]: oops

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt (hunk 9 above)
Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:9]: permBlacklist is unmutable, as Aliucord won't allow this ever.
the servBlacklist is mutable, this is what we modify with commands

Comment (X1nto) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:9]: Understandable, but what did Vendetta do?

Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:9]: they didnt allow invisible characters too so

**Comment by OmegaSunkey** [comment]
Almost done. 
If you wonder why is it now a function its because the previous way made it return null because when NitroSpoof loads it is null.
Now i'll fix the last things.

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt (hunk 9 above)
Comment (X1nto) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt:9]: Ah, understood.

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt (hunk 8 above)
Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/Util.kt]: Done.

**Comment by OmegaSunkey** [comment]
Should be done. Can you do the last review?

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt [hunk 10]
```diff
@@ -37,7 +39,19 @@ class PluginSettings(
             }
         }
 
+        val emptyToggle = Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, "Enable Empty Character", "This allows you to embed your emote without the URL being present. Some servers do not allow empty character.").apply {
+		this.isChecked = settingsAPI.getBool("emptyChar", false)
+		this.setOnCheckedListener {
+			c: Boolean? -> if (PluginManager.isPluginEnabled("MoreHighlight")) settingsAPI.setBool("emptyChar", c!!) else {
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:45]: I honestly don't understand this style of lambda declarations. Just don't.

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt [hunk 11]
```diff
@@ -37,7 +39,19 @@ class PluginSettings(
...
         }
 
+        val emptyToggle = Utils.createCheckedSetting(context, CheckedSetting.ViewType.SWITCH, "Enable Empty Character", "This allows you to embed your emote without the URL being present. Some servers do not allow empty character.").apply {
+		this.isChecked = settingsAPI.getBool("emptyChar", false)
+		this.setOnCheckedListener {
+			c: Boolean? -> if (PluginManager.isPluginEnabled("MoreHighlight")) settingsAPI.setBool("emptyChar", c!!) else { 
+			Utils.showToast("Please install MoreHighlight for this to work.") 
+			settingsAPI.setBool("emptyChar", false)
+        		}
+        	}
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:49]: Fix indentation

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt [hunk 12]
```diff
@@ -1,105 +1,149 @@
 package com.xinto.aliuplugins
 
+// import java.io.File
```
Comment (X1nto) [inline plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/NitroSpoof.kt:3]: Why is this commented out? Just remove the import.

**Review by X1nto — CHANGES_REQUESTED** [review]

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt (hunk 10 above)
Comment (OmegaSunkey) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:45]: You mean the name of the argument or ?

Comment (X1nto) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:45]: No, I mean placing the lambda arguments on a newline. The convention is the following:
```kt
this.setOnCheckedListener { c: Boolean -> 

}
```

**Comment by X1nto** [comment]
What's up with the indentation? Sometimes it's 2 spaces, sometimes it's 4. I use 4 spaces so please stick to that.

File: plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt (hunk 11 above)
Comment (X1nto) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:49]: You didn't fix bracket indentation.

Comment (X1nto) [reply plugins/NitroSpoof/src/main/java/com/xinto/aliuplugins/nitrospoof/PluginSettings.kt:49]: Also, I see you switched to tabs from spaces. Use spaces.

**Comment by Xynonners** [comment]
uh, is PR ever going to get merged?

**Comment by X1nto** [comment]
> uh, is PR ever going to get merged?

When requested changes are made and conflicts are solved, yes.

**Comment by OmegaSunkey** [comment]
> uh, is PR ever going to get merged?

i dont even think this should be added now. it would be better to do what newer plugins do (adding emoji name to markdown)

**Comment by Xynonners** [comment]
> > uh, is PR ever going to get merged?
> 
> i dont even think this should be added now. it would be better to do what newer plugins do (adding emoji name to markdown)

wouldn't it be the same in code to hyperlink the emoji name vs an empty character such as: https://unicode-explorer.com/c/2065

**Pull request closed without merging**